packages/server/lib/cypress.js

```diff
@@ -164,19 +164,23 @@ module.exports = {
       options.headed = false
     }
 
+    const electronApp = require('./util/electron-app')
+
     if (options.runProject && !options.headed) {
       debug('scaling electron app in headless mode')
       // scale the electron browser window
       // to force retina screens to not
       // upsample their images when offscreen
       // rendering
-      require('./util/electron-app').scale()
+      electronApp.scale()
     }
 
+    electronApp.setScopeMemoryCachePerContext()
```

Comment:
```suggestion
    // control memory caching per execution context so that font flooding does not occur: https://github.com/cypress-io/cypress/issues/28215
    electronApp.setScopeMemoryCachePerContext()
```